pkg/controller/backupentry/actuator.go

```diff
@@ -0,0 +1,132 @@
+	Delete(context.Context, *extensionsv1alpha1.BackupEntry) error
+	// GetETCDSecretData return the updated the secret data as per provider requirement.
+	GetETCDSecretData(context.Context, *extensionsv1alpha1.BackupEntry, map[string][]byte) (map[string][]byte, error)
+}
+
+type actuator struct {
+	providerActuator ProviderActuator
+	client           client.Client
+	logger           logr.Logger
+}
+
+// InjectClient injects the given client into the valuesProvider.
+func (a *actuator) InjectClient(client client.Client) error {
+	a.client = client
+	return nil
+}
+
+// InjectFunc enables injecting Kubernetes dependencies into actuator's dependencies.
+func (a *actuator) InjectFunc(f inject.Func) error {
+	return f(a.providerActuator)
+}
+
+// NewActuator creates a new Actuator that updates the status of the handled BackupBucket resources.
+func NewActuator(provideActuator ProviderActuator, logger logr.Logger) Actuator {
+	return &actuator{
+		logger:           logger,
+		providerActuator: provideActuator,
+	}
+}
+
+// Reconcile reconciles the update of a BackupEntry
+func (a *actuator) Reconcile(ctx context.Context, be *extensionsv1alpha1.BackupEntry) error {
+	return a.deployEtcdBackupSecret(ctx, be)
+}
+
+func (a *actuator) deployEtcdBackupSecret(ctx context.Context, be *extensionsv1alpha1.BackupEntry) error {
+	shootTechnicalID, _ := extractShootDetailsFromBackupEntryName(be.Name)
+	namespace := &corev1.Namespace{}
+	if err := a.client.Get(ctx, kutil.Key(shootTechnicalID), namespace); err != nil {
+		if apierrors.IsNotFound(err) {
+			a.logger.Info("SeedNamespace for shoot not found. Avoiding etcd backup secret deployment")
+			return nil
+		}
+		a.logger.Error(err, "failed to get seed namespace")
+		return err
+	}
+
+	backupSecret, err := controller.GetSecretByReference(ctx, a.client, &be.Spec.SecretRef)
+	if err != nil {
+		a.logger.Error(err, "failed to read backup extension secret")
+		return err
+	}
+
+	backupSecret.Data[backupBucketName] = []byte(be.Spec.BucketName)
+	etcdSecretData, err := a.providerActuator.GetETCDSecretData(ctx, be, backupSecret.Data)
+	if err != nil {
+		return err
+	}
+	etcdSecret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      EtcdBackupSecretName,
+			Namespace: shootTechnicalID,
+		},
+		Data: etcdSecretData,
+	}
+	return controller.CreateOrUpdate(ctx, a.client, etcdSecret, nil)
+}
+
+// Delete deletes the BackupEntry
+func (a *actuator) Delete(ctx context.Context, be *extensionsv1alpha1.BackupEntry) error {
+	return a.providerActuator.Delete(ctx, be)
+}
+
+// extractShootDetailsFromBackupEntryName returns Shoot resource technicalID its UID from provided <backupEntryName>.
+func extractShootDetailsFromBackupEntryName(backupEntryName string) (shootTechnicalID, shootUID string) {
+	tokens := strings.Split(backupEntryName, "--")
+	shootUID = tokens[len(tokens)-1]
+	shootTechnicalID = strings.TrimSuffix(backupEntryName, shootUID)
+	shootTechnicalID = strings.TrimSuffix(shootTechnicalID, "--")
```

Comment:
Why do you call `TrimSuffix` twice?

```suggestion
	shootTechnicalID = strings.TrimSuffix(shootTechnicalID, "--" + shootUID)
```